ONSdigital/eq-author-app
# EAR-677-validate-or-on-checkboxes-in-logic

### What is the context of this PR?

This pull request improves upon existing validation for mutually exclusive checkboxes in routing. The backend has been streamlined and bugs removed, whilst the front end has been enhanced by the addition of red borders around included fields to help the user notice what needs to be changed.

### How to review 

Using the ACs on the Jira ticket, ensure that validation states can be correctly reached and are are not seen in other circumstances where an error being shown is incorrect.

* Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
    * ensure it can be opened in Author;
    * then, ensure it can be viewed in Runner by pressing the **view survey** button
* Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Jira ticket for this task into the next stage of the process
